sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtensionTest.java

```diff
@@ -0,0 +1,74 @@
+package org.sdase.commons.server.mongo.testing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sdase.commons.server.mongo.testing.MongoDbClassExtension.OVERRIDE_MONGODB_CONNECTION_STRING_ENV_NAME;
+
+import com.mongodb.MongoClient;
+import com.mongodb.client.FindIterable;
+import org.bson.Document;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.sdase.commons.server.testing.Environment;
+
+class UseExistingMongoDbClassExtensionTest {
+
+  @RegisterExtension
+  static final MongoDb EXTERNAL_DB =
+      MongoDbClassExtension.builder()
+          .withDatabase("testDb")
+          .withUsername("testuser")
+          .withPassword("testpassword")
+          .build();
+
+  private MongoDbClassExtension useExistingMongoDbClassExtension;
+
+  @BeforeEach
+  public void initUseExistingMongoDbRule() {
+
+    String MONGODB_CONNECTION_STRING =
+        "mongodb://"
+            + EXTERNAL_DB.getUsername()
+            + ":"
+            + EXTERNAL_DB.getPassword()
+            + "@"
+            + EXTERNAL_DB.getHosts()
+            + "/"
+            + EXTERNAL_DB.getDatabase()
+            + "?"
+            + EXTERNAL_DB.getOptions();
+    System.getenv().get(OVERRIDE_MONGODB_CONNECTION_STRING_ENV_NAME);
```

Comment:
I think you wanted to store the old value to be able to restore it after the test, right? :)

Reply:
oh yes